### PR TITLE
S/Kademlia package tests.

### PR DESCRIPTION
**Pull request by doug-perlin**

### CODE REVIEW

**Comment by doug-perlin** [comment]
This branch should be okay to go into v2 for now.
Mocking out peer and node may be needed for other things to tested, will put that in a different PR.

File: skademlia/table_test.go [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+
+	testee = []ID{}
+	for _, peer := range FindClosestPeers(routingTable, nodes[4].Hash(), 2) {
+		testee = append(testee, peer.(ID))
+	}
+	if len(testee) != 2 {
+		t.Fatalf("findclosestpeers() error, size of return should be 2, but found %d", len(testee))
+	}
+	answerKeys = []int{4, 5}
+	// TODO: should be {4, 2} not {4, 5}
```
Comment (iwasaki-kenta) [inline skademlia/table_test.go]: Does this still need to be addressed?

**Comment by doug-perlin** [comment]
One thing strange when going through this was what happened to the skademlia C1/C2 puzzle for the ID?

**Comment by iwasaki-kenta** [comment]
> One thing strange when going through this was what happened to the skademlia C1/C2 puzzle for the ID?

That still needs to be implemented for the time being; I was readily porting Kademlia first before S/Kademlia and have yet to implement the PoW-based C1/C2 puzzle just yet.

Would you be able to implement it quickly within `skademlia.ID`?

**Comment by doug-perlin** [comment]
The puzzle and check shouldn't be much more.

**Pull request merged**